test/core/event_engine/test_suite/tests/client_test.cc

```diff
@@ -293,5 +302,100 @@ TEST_F(EventEngineClientTest, MultipleIPv6ConnectionsToOneOracleListenerTest) {
   server_endpoint.reset();
 }
 
+// Create a connection using the test EventEngine to a listener created by the
+// test EventEngine and exchange bi-di data over the connection. Each endpoint
+// gets reset as soon as the write is done. This test checks that EventEngine
+// implementations handle lifetimes around endpoints correctly.
+
+TEST_F(EventEngineClientTest, WriteEventCallbackEndpointValidityTest) {
+  grpc_core::ExecCtx ctx;
+  std::shared_ptr<EventEngine> test_ee(this->NewEventEngine());
+  auto memory_quota = std::make_unique<grpc_core::MemoryQuota>("bar");
+  std::string target_addr = absl::StrCat(
+      "ipv6:[::1]:", std::to_string(grpc_pick_unused_port_or_die()));
+  auto resolved_addr = URIToResolvedAddress(target_addr);
+  CHECK_OK(resolved_addr);
+  std::unique_ptr<EventEngine::Endpoint> client_endpoint;
+  std::unique_ptr<EventEngine::Endpoint> server_endpoint;
+  std::unique_ptr<grpc_core::Notification> server_signal;
+
+  Listener::AcceptCallback accept_cb =
+      [&server_endpoint, &server_signal](
+          std::unique_ptr<Endpoint> ep,
+          grpc_core::MemoryAllocator /*memory_allocator*/) {
+        server_endpoint = std::move(ep);
+        server_signal->Notify();
+      };
+
+  grpc_core::ChannelArgs args;
+  auto quota = grpc_core::ResourceQuota::Default();
+  args = args.Set(GRPC_ARG_RESOURCE_QUOTA, quota);
+  ChannelArgsEndpointConfig config(args);
+  auto listener = *test_ee->CreateListener(
+      std::move(accept_cb),
+      [](absl::Status status) {
+        ASSERT_TRUE(status.ok()) << status.ToString();
+      },
+      config, std::make_unique<grpc_core::MemoryQuota>("foo"));
+
+  ASSERT_TRUE(listener->Bind(*resolved_addr).ok());
+  ASSERT_TRUE(listener->Start().ok());
+
+  for (int i = 0; i < 100; ++i) {
```

Comment:
nit: try to avoid magic numbers, and consider declaring a `constexpr size_t n_iterations = 100;` for it.

Reply:
done